src/spatialdata/dataloader/datasets.py

```diff
@@ -114,9 +117,18 @@ def __getitem__(self, idx: int) -> tuple[SpatialImage, str, int]:
         if isinstance(regions, GeoDataFrame):
             dims = get_axes_names(regions)
             region = regions.iloc[region_index]
-            # the function coords.xy is just accessing _coords, and wrapping it with extra information, so we access
-            # it directly
-            centroid = np.atleast_2d(region.geometry.coords._coords[0])
+            shape = regions.geometry.iloc[0]
+            if isinstance(shape, Polygon):
+                centroid = np.atleast_2d(region.geometry.centroid.coords._coords[0])
+            elif isinstance(shape, MultiPolygon):
+                raise NotImplementedError("MultiPolygon not supported yet")
+            elif isinstance(shape, Point):
+                # the function coords.xy is just accessing _coords, and wrapping it with extra information,
+                # so we access it directly
+                centroid = np.atleast_2d(region.geometry.coords._coords[0])
```

Comment:
Do we have a test to cover this to catch if they change the private API?

Reply:
Good point! Seems to be already broken. I am replacing this with public APIs and adding some tests (so far the code is not tested).

Reply:
Turned out the private APIs were still working (and did something wrong with the tests). Anyway, I removed them and now we are using public APIs.